src/firefly/java/edu/caltech/ipac/firefly/server/visualize/VisServerOps.java

```diff
@@ -69,6 +70,10 @@ public class VisServerOps {
     static {
         VisContext.init();
     }
+    private static final CacheKey BYTE_STRETCH_SEMAPHORE = new StringKey("BYTE_STRETCH_SEMAPHORE");
+    private static final int MAX_SEMAPHORE_WAIT_SECS = 360;
+    private static final int BYTE_STRETCH_MAX_ACCESS_CNT = 1;
+    private static final String BYTE_STRETCH_SEMAPHORE_KEY = "BYTE_STRETCH_SEMAPHORE";
```

Comment:
Are you expecting there will be cases where BYTE_STRETCH_MAX_ACCESS_CNT might need to be greater than 1?
Is that why you're opting for a Semaphore instead of a Lock (e.g., ReentrantLock)?

Reply:
This allows for making it 2 or 3 in the future. It is more flexible. There is nothing about the function that requires a lock, it is just a way to slow traffic so a semaphore make the most sense.